### PR TITLE
Add Liveness Probe for EventListener Deployment

### PR DESCRIPTION
**Pull request by khrm**
Add a Liveness Probe for EventListener Deployment at URL /live
Fixes: https://github.com/tektoncd/triggers/issues/276

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
Add a Liveness Probe for EventListener Deployment at URL /live
```

### CODE REVIEW

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go [hunk 1]
```diff
@@ -541,6 +553,18 @@ func TestReconcile(t *testing.T) {
 							ContainerPort: int32(*ElPort),
 							Protocol:      corev1.ProtocolTCP,
 						}},
+						LivenessProbe: &corev1.Probe{
+							Handler: corev1.Handler{
+								HTTPGet: &corev1.HTTPGetAction{
+									Path:   "/live",
+									Scheme: corev1.URISchemeHTTP,
+									Port:   intstr.FromInt((*ElPort)),
+									Host:   "localhost",
```
Comment (ncskier) [inline pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go]: I think we need to remove this `localhost` line.

**Comment by khrm** [comment]
@ncskier I think that was due to localhost.

**Comment by dibyom** [comment]
@khrm Can we add some docs for this feature?

**Comment by khrm** [comment]
OK. I would add it.

**Pull request merged**